src/doc/rustdoc/src/the-doc-attribute.md

```diff
@@ -0,0 +1,108 @@
+# The `#[doc]` attribute
+
+The `#[doc]` attribute lets you control various aspects of how `rustdoc` does
+its job. 
+
+The most basic job of `#[doc]` is to be the way that the text of the documentation
+is handled. That is, `///` is syntax sugar for `#[doc]`. This means that these two
+are the same:
+
+```rust,ignore
+/// This is a doc comment.
+#[doc = "This is a doc comment."]
+```
+
+In most cases, `///` is easier to use than `#[doc]`. One case where the latter is easier is
+when generating documentation in macros; the `collapse-docs` pass will combine multiple
+`#[doc]` attributes into a single doc comment, letting you generate code like this:
+
+```rust,ignore
+#[doc = "This is"]
+#[doc = " a "]
+#[doc = "doc comment"]
```

Comment:
Except `collapse-docs` will put these all on multiple lines. It's not necessarily the same as `///This is a doc comment`, all on one line.